## Тема: Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона.

In [1]:
import numpy as np
from math import factorial

# функция расчета количества сочетаний
def combinations(k, n):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Можно применить формулу Бернулли $P_n = C_n^k p^k q^{n-k}$  

In [2]:
n = 100
k = 85
p = 0.8
q = 1 - p
P_n = np.round(combinations(k, n) * (p**k) * (q**(n-k)), 3)
P_n

0.048

**Ответ:** Вероятность того, что стрелок попадет в цель ровно 85 раз равно 0.048

### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.   

### Какова вероятность, что ни одна из них не перегорит в первый день? 

Можно применить формулу Пуассона $P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$

In [3]:
m1 = 0
n = 5000
p = 0.0004
lambda_ = n * p

P1 = np.round((lambda_**m1 / np.math.factorial(m1)) * np.exp(-lambda_), 4)
P1

0.1353

### Какова вероятность, что перегорят ровно две? 

In [4]:
m2 = 2
n = 5000
p = 0.0004
lambda_ = n * p

P2 = np.round((lambda_**m2 / np.math.factorial(m2)) * np.exp(-lambda_), 4)
P2

0.2707

**Ответ:** вероятность, что ни одна лампочка не перегорит равна 0,1353 или 13%
           вероятность, что перегорят ровно две лампочки равна 0,2707 или 27%

### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?  

Можно применить формулу Бернулли $P_n = C_n^k p^k q^{n-k}$ 

In [15]:
p = 1/2
q = 1/2
k = 70  
n = 144

P_n = np.round(combinations(k, n) * (p**k) * (q**(n-k)), 4)
P_n

0.0628

**Ответ:** вероятность попадания равна 0.0628 или 6%

### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. 

### Какова вероятность того, что все мячи белые?  

Можно использовать формулу **условной вероятности**

In [48]:
# вероятность извлечения 2 белых мячей из 1 и из 2 ящика
С = 7/10 * 6/9 * 9/11 * 8/10
С

0.3054545454545454

**Ответ:** вероятность извлечения 4 белых мячей равна 0.3054

### Какова вероятность того, что ровно два мяча белые?

Событие состоит из 3 случаев: 
1. 2 белых и 0 черных из первого ящика, 0 белых и 2 черных из второго
2. 1 белый и 1 черный из первого ящика, 1 белый и 1 черный из второго
3. 0 белых и 2 черных из первого ящика, 2 белых и 0 черных из второго

In [40]:
p1 = np.round(combinations(2, 7)/combinations(2, 10)*combinations(2, 2)/combinations(2, 11), 4)
p1

0.0085

In [43]:
p2 = np.round(combinations(1, 7)*combinations(1, 3)/combinations(2, 10)*combinations(1, 9)*combinations(1, 2)/combinations(2, 11), 4)
p2

0.1527

In [42]:
p3 = np.round(combinations(2, 3)/combinations(2, 10)*combinations(2, 9)/combinations(2, 11), 4)
p3

0.0436

In [44]:
p = p1 + p2 + p3
p

0.2048

**Ответ:** вероятность извлечения ровно двух мячей равна 0.2048

### Какова вероятность того, что хотя бы один мяч белый?

Можно использовать формулу противоположного события $P(A) = 1 - P(\overline{A})$.

Для расчета вероятности противоположного события **А_** можно использовать формулу **условной вероятности**. 

In [20]:
A_ = np.round(3/10 * 2/9 * 2/11 * 1/10, 4)
A_

0.0012

In [21]:
A = (1 - A_)
A

0.9988

**Ответ:** вероятность извлечения хотя бы одного белого мяча равна 0.9964 